python/local/RGroupReplacement.yaml

```diff
@@ -0,0 +1,84 @@
+id: 0227312a-b83e-9777-d285-196c05263676
+name: R Group Replacement
+description: 'Uses the Top 500 R Group replacement data of Bajorath et al. to produce analogues of the input molecules. '
+category: Chemistry
+version: 1.0.0
+serviceName: RGroupReplacement
+serviceUri: glysade.python
+executorId: Glysade.CPythonDataFxn
+inputFields:
+- control:
+    id: structureColumn
+    label: Select structure column
+    type: columnselect
+    filters:
+    - dataType: string
+      contentType:
+      - chemical/x-mdl-molfile
+      - chemical/x-mdl-molfile-v3000
+      - chemical/x-smiles
+      - chemical/x-daylight-smiles
+    - dataType: binary
+      contentType:
+      - chemical/x-mdl-molfile
+      - chemical/x-mdl-molfile-v3000
+    validationRules:
+    - type: required
+      message: Must select column of 2D structures
+  request:
+    id: structureColumn
+    dataType: string
+    selectorType: column
+- control:
+    id: useLayer1
+    label: Use Layer 1 replacements
+    type: checkbox
+    tooltip: The direct replacements
+  request:
+    id: useLayer1
+    dataType: boolean
+    data: !!bool true
+- control:
+    id: useLayer2
+    label: User Layer 2 replacements
+    type: checkbox
+    tooltip: Layer 2 replacements are nearest neighbours of their parent Layer 1 replacement.
+  request:
+    id: useLayer2
+    dataType: boolean
+    data: !!bool false
```

Comment:
The sketcher is set for reactions, but we want a substructure query.

Reply:
I have attempted to fix this.  It now uses a SMARTS query correctly to generate 72 products from the Chembl set, but what I think is the corresponding MOLfile query gives nothing.  I am wondering if this is because it's not being built as a query molecule, though I have manually changed the aromatic bond types to type 4 rather than the kekule form.  I'm not getting the error you saw, but also not getting any hits.  I am at a loss, I'm afraid.  I expect I'm not passing a type in correctly somewhere.  A chat would no doubt solve it quickest.

Reply:
Happy to chat, but ATM I'm also stumped.  The only issue I've had with queries in mol blocks is that sanitization may not work for partial ring systems.  I'll play with things and see if I can figure out what is happening.